src/python/pants/backend/python/goals/run_python_source.py

```diff
@@ -48,6 +51,18 @@ def should_use_sandbox(self, python_setup: PythonSetup) -> bool:
             return python_setup.default_run_goal_use_sandbox
         return self._run_goal_use_sandbox.value
 
+    def _executable_main(self) -> Optional[Executable]:
+        source = PurePath(self.source.value)
+        source_name = source.stem if source.suffix == ".py" else source.name
+        if not all(part.isidentifier() for part in source_name.split(".")):
+            # If the python source is not importable (python modules can't be named with '-'),
+            # then it must be an executable script.
+            executable = Executable(self.source.value)
```

Comment:
Oops. `self.source.value` is relative to the BUILD file, but Executable needs to be relative to the workspace root, similar to the logic in `PexExecutableField`.
```
executable = Executable(os.path.join(self.address.spec_path, self.source.value).lstrip(os.path.sep))
```

Reply:
The integration tests are only testing running a `pex_binary`, not `python_source`.
So, we need to add another parameter to this test to use `app.py` or `my-app.py` so it uses the Executable main:
https://github.com/pantsbuild/pants/blob/main/src/python/pants/backend/python/goals/run_python_source_integration_test.py#L114-L137